src/games.py

```diff
@@ -0,0 +1,140 @@
+import db, utils, config
+from datetime import datetime, timedelta, timezone
+import asyncio, random
+from config import ADMIN_ACCESS
+
+ANNOUNCE_MESSAGES = [
+    "Oh ho ho, what are all these free games I've found?",
+    "Wow, I still remember how delicious the Luau soup was. Here are some free games:",
+    "Here are today's free games!"
+]
+
+"""
+Game Timer
+
+Manages sending out game announcements at a regular interval.
+Can only handle one announcement channel for one server.
+"""
+class GameTimer:
+    def start(self, channel):
+        self._channel = channel
+
+        asyncio.create_task(self._announce_games())
+
+    async def _announce_games(self):
+        while True:
+            await self._wait_until_next_announcement()
+
+            games = db.get_games()
+
+            if len(games) != 0:
+                # generate message
+                formatted_games = "\n".join(games)
+                announcement = random.choice(ANNOUNCE_MESSAGES)
+                message = f"{announcement}\n\n{formatted_games}"
+
+                # send message
+                await self._channel.send(message)
+
+                # clear games
+                db.clear_games()
+
+    @staticmethod
+    async def _wait_until_next_announcement():
+        next_announcement = get_delta_to_next_announcement()
+
+        await asyncio.sleep(next_announcement.total_seconds())
+
+
+"""
+Requires admin
+
+Wrapper function for commands that has them do nothing if the author doesn't have the admin role
+"""
+def requires_admin(func):
```

Comment:
Does `requires_admin` need to be async?

This is dope, we should reuse this for the rest of the admin functions

Reply:
No - from what I can tell decorators are equivalent to

```python
async def add_game(message):
   ...
add_game = requires_admin(add_game)
```

So the returned function needs to be async because it wraps async functions, but the decorator itself is just a regular function.

I can move it to another file and also use it in `commands.py` - `utils.py` perhaps?

Reply:
Yeah, I think `utils.py` would probably be the most ideal place for it.

Reply:
Made the change. I needed to modify the wrapper to support functions with more than a message argument

e.g. `CustomCommands::define_cmd` has both `self` and `message` arguments.

So I changed the wrapper to assume that the message is the last non-keyword argument. Not sure if there's a better way to do that.